pymodbus/server/simulator/http_server.py

```diff
@@ -31,6 +32,9 @@
 )
 
 
+if TYPE_CHECKING:  # type checkers do not understand the Raise RuntimeError in __init__()
```

Comment:
You invented a different method for importing conditionally (pyserial), we should use the same strategy here.

Reply:
No, look at the whole import area instead of just the changes.

You remember correctly that I wrapped the conditional imports (for both `pyserial` and `aiohttp` so they can `raise RuntimeError` and warn the user.    This remained unchanged.

The type checkers are not smart enough to realize that the `RuntimeError` will prevent any further code execution.  So they throw "warnings" that later usages of `web` (or `serial`) may be unbound (in the case the import failed).
This new code only serves to "convince" type checkers the import succeeded.   It will never execute at runtime. (`TYPE_CHECKING == False`).

Reply:
Now I see you have done the same in serial...I know why I do not like mypy and companions.

Anyhow it is confusing to have the same import in 2 places can't you do "if type checking else" and have the it as one block ?

Reply:
> Anyhow it is confusing to have the same import in 2 places can't you do "if type checking else" and have the it as one block ?

How about this?
```python
try:
    from aiohttp import web

    AIOHTTP_MISSING = False
except ImportError:
    AIOHTTP_MISSING = True
    if TYPE_CHECKING:  # always False at runtime
        # type checkers do not understand the Raise RuntimeError in __init__()
        from aiohttp import web
```